<launch>

   <arg name="joystick_type" default="xbox" />
   <arg name="output_message_type" default="romea_mobile_base_msgs::msg::SkidSteeringCommand"/>
   <arg name="output_message_priority" default="0"/>
   <arg name="slow_mode_maximal_linear_speed"/>
   <arg name="turbo_mode_maximal_linear_speed" />
   <arg name="slow_mode_angular_linear_speed"/>
   <arg name="turbo_mode_angular_linear_speed" />

  <node pkg="romea_teleop" name="teleop_node" exec="skid_steering_teleop_node" output="screen">
    <param name="joystick.type" value="$(var joystick_type)"/>
    <param from="$(find-pkg-share romea_teleop)/config/$(var joystick_type)_skid_steering_remappings.yaml"/>
    <param name="cmd_output.type" value="$(var output_message_type)"/>
    <param name="cmd_output.priority" value="$(var output_message_priority)"/>
    <param name="cmd_range.maximal_linear_speed.slow_mode" value="$(var slow_mode_maximal_linear_speed)"/>
    <param name="cmd_range.maximal_linear_speed.turbo_mode" value="$(var turbo_mode_maximal_linear_speed)"/>
    <param name="cmd_range.maximal_angular_speed.slow_mode" value="$(var slow_mode_angular_linear_speed)"/>
    <param name="cmd_range.maximal_angular_speed.turbo_mode" value="$(var turbo_mode_angular_linear_speed)"/>
  </node>

</launch>
